frontend/helpers/redirect-helper.ts

```diff
@@ -0,0 +1,18 @@
+import { IncomingMessage } from "http";
+
+export function getSubDomainRedirect(request: IncomingMessage, redirectPath: string) {
+   const regex = new RegExp(process.env.SUBDOMAIN_REDIRECT_UNLESS_REGEX || '^');
+   const host = request.headers.host;
+   const shouldRedirect =  host ? !host.match(regex) : false;
+
+   if (shouldRedirect) {
+      const url = new URL(process.env.NEXT_PUBLIC_IFIXIT_ORIGIN || "");
+      url.pathname = redirectPath;
```

Comment:
It might be cute to use https://nodejs.org/dist/latest-v15.x/docs/api/http.html#http_message_url, instead of `redirectPath`, because it looks like that should basically always do exactly what we want: have just the path part of the URL which was used to get here in the first place.

Free bonus, it lets us get rid of `redirectPath`, which simplifies all the calls to this function.
```suggestion
      url.pathname = request.url;
```